### PR TITLE
Log face prefactor

### PR DESCRIPTION
**Pull request by peterMuriuki**
part of #13.

- Refactor prop typings: fixes optional props(due to defaultProps) are not scriggly-lined as errors
- Refactor error reporting in the LogFace container:
  - to this end Installed react-hot-toast and added an ErrorPage to be shown for handled unrecoverable errors.

### CODE REVIEW

File: src/components/ErrorPage/index.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+/** Error page component shown in pages that experience unrecoverable errors */
+const ErrorPage = (props: ErrorPageProps) => {
+    const { title, message } = props;
+    return (
+        <div className="container">
+            <div className="row">
+                <div className="col-md-12">
+                    <div className="error-template">
+                        <h1>An error occurred</h1>
```
Comment (moshthepitt) [inline src/components/ErrorPage/index.tsx]: Do we care a lot about i18n right now?

**Review by moshthepitt — DISMISSED** [review]

File: src/components/ErrorPage/index.tsx (hunk 1 above)
Comment (peterMuriuki) [reply src/components/ErrorPage/index.tsx]: will fix

**Review by moshthepitt — APPROVED** [review]

File: src/containers/LogFace/index.tsx [hunk 2]
```diff
@@ -294,15 +325,22 @@ export const LogFace = ({
         totalRecords: filteredData.length,
     };
 
+    if (broken) {
+        return <ErrorPage title={error?.name} message={error?.message} />; // TODO
```
Comment (machariamuguku) [inline src/containers/LogFace/index.tsx:329]: ErrorPage expects title, and message to be a string and not optional, there's potential to be undefined here?

File: src/containers/LogFace/index.tsx [hunk 3]
```diff
@@ -294,15 +325,22 @@ export const LogFace = ({
...
     return (
         <div className="logface-content">
             <div>
                 <h2 id="logface_title">{t(`Log Face - ${module}`)}</h2>
             </div>
             <div className="filter-panel">
                 <div className="filters">
-                    {/*tslint:disable-next-line: jsx-no-lambda no-empty*/}
-                    <Formik initialValues={{}} onSubmit={() => {}}>
+                    <Formik initialValues={{}} onSubmit={() => void 0}>
```
Comment (machariamuguku) [inline src/containers/LogFace/index.tsx:343]: do we really need formik for this?

**Review by machariamuguku — APPROVED** [review]

**Pull request merged**